sql/src/main/java/org/apache/druid/sql/calcite/schema/SystemSchema.java

```diff
@@ -241,7 +250,7 @@ public Map<String, Table> getTableMap()
   /**
    * This table contains row per segment from metadata store as well as served segments.
    */
-  static class SegmentsTable extends AbstractTable implements ScannableTable
+  static class SegmentsTable extends AbstractTable implements ScannableTable, ProjectableFilterableTable
```

Comment:
Do we still need to implement `ScannableTable`? I'd think the new `scan(DataContext, List<RexNode>, int[])` signature would come with empty filters and null projects if it's not present in the query

Reply:
It seems it doesn't. I removed `ScannableTable`.